File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        prod, zero_count = 1, 0

        for num in nums:
            if num == 0:
                zero_count += 1
                if zero_count > 1:
                    return [0] * len(nums)
            else:
                prod *= num
        
        res = [0] * len(nums)

        for i, num in enumerate(nums):
            if zero_count < 1:
                res[i] = prod // nums[i]
            else:
                if num == 0:
                    res[i] = prod
                else:
                    res[i] = 0
        return res

File: Data Structures & Algorithms/counting-bits/submission-0.java
class Solution {
    public int[] countBits(int n) {
        int[] result = new int[n + 1];
        result[0] = 0;

        for (int i = 1; i <= n; i++) {
            int sum = 0;
            int num = i;
            while (num != 0) {
                sum += 1;
                num &= num - 1;
            }
            result[i] = sum;
        }
        return result;
    }
}
